ckanext/etsin/mappers/datacite.py

```diff
@@ -0,0 +1,257 @@
+            break
+
+    # Map publisher
+    publisher = xml.find('.//publisher').text
+    package_dict['publisher'] = [{'name': publisher}]
+
+    # Map publication year
+    publication_year = xml.find('.//publicationYear').text
+    package_dict['issued'] = publication_year
+
+    # Map subject
+    package_dict['keyword'] = []
+    package_dict['theme'] = []
+    for subject in xml.findall('.//subject'):
+        subject_scheme = subject.get('subjectScheme')
+        scheme_URI = subject.get('schemeURI')
+        value_URI = subject.get('valueURI')
+        if subject_scheme is None and scheme_URI is None:
+            package_dict['keyword'].append(subject.text)
+        elif subject_scheme == "YSO" or "finto.fi/yso" in scheme_URI:
+            if value_URI is not None:
+                package_dict['theme'].append({'identifier': value_URI})
+            elif is_uri(subject.text):
+                package_dict['theme'].append({'identifier': subject.text})
+
+    # Map contributor
+    package_dict['contributor'] = []
+    package_dict['curator'] = []
+    package_dict['rights_holder'] = []
+    for contributor in xml.findall('.//contributor'):
+        contributor_type = contributor.find('.//contributorType').text
+        if contributor_type in ["DataCollector", "DataCurator", "DataManager", "Editor", "Producer", "ProjectLeader", "ProjectMember", "Researcher", "ResearchGroup", "Supervisor"]:
+            metax_contributor_type = "contributor"
+        elif contributor_type == "Distributor":
+            metax_contributor_type = "publisher"
+        elif contributor_type == "ContactPerson":
+            metax_contributor_type = "curator"
+        elif contributor_type == "RightsHolder":
+            metax_contributor_type = "rights_holder"
+        else:
+            continue
+        person = _get_person(contributor)
+        if not Person:
+            continue
+        package_dict[metax_contributor_type].append(person)
+
+    # Map date
+    package_dict['provenance'] = []
+    for date in xml.findall('.//date'):
+        date_type = date.get('dateType')
+        if date_type in ['todo: find correct reference data']:  # TODO
+            package_dict['provenance'].append({
+                'temporal': {
+                    'start_date': date.text,
+                    'end_date': date.text
+                },
+                'type': {
+                    'pref_label': dateType,
+                    'identifier': 'TODO'  # TODO
+                }
+            })
+
+    # Map alternate identifier
+    package_dict['other_identifier'] = []
+    for alternate_identifier in xml.findall('.//alternateIdentifier'):
+        alternate_identifier_type = alternate_identifier.get(
+            'alternateIdentifierType')
+        if alternate_identifier_type == "URL":
+            package_dict['other_identifier'].append({
+                'notation': alternate_identifier.text,
+                'type': alternate_identifierType,
```

Comment:
Type is an object, which should have an identifier derived from identifier_type reference data.

Reply:
Identifier_type.json currently has an entry with ID "url". Should we check every time that the entry is still in reference data?

Reply:
Also: I fixed the typo in variable name.